docs/image_auth.md

```diff
@@ -0,0 +1,95 @@
+
+## Work Process
+
+This section will clarify the authorization and authentication process of a registry.
+First let us make some assumptions to help.
+
+### Basic Model
+
+- User `Alice` wants to deploy an image `busybox` inside a confidential VM, s.t. pod. Let's say the host of the confidential VM is `Bob`, who can represent CSP.
+- The image is from a private registry `private-registry.org` which needs credential to login.
+- The image is pulled inside the VM to prevent `Bob` from knowing the credential.
+- `Alice` has the credential of this registry, s.t. `username:Alice`, `password:pswd`
+- The pod of `Bob` side to deploy confidential container is exclusive by `Alice`.
+
+### Authorization
+
+`Alice` should first edit an `auth.json` following [the format](https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md#format). In this story, the `auth.json` may look like
+
+```json
+{
+	"auths": {
+		"private-registry.org": {
+			"auth": "QWxpY2U6cHN3ZAo="
+		},
+	}
+}
+```
+
+> **Warning**: Because upstream `oci_distribution` does not support credential helpers, so we just support auth token here.
+
+Here `QWxpY2U6cHN3ZAo=` is the base64-encoded `Alice:pswd`.
+All the credentials have a common format that `<username>:<password>`.
+
+All above happens on the `Alice`'s side, so it is exactly safe and unknown to `Bob`.
+
+### Authentication
+
+When `Alice` wants to deploy the image `private-registry.org/busybox` inside the confidential VM of `Bob`'s side, `image-rs` needs to pull the image.
+
+1. Firstly, it checks whether there is an file `auth.json`, if yes, do step 3.
+2. Fetch the `auth.json` via `GetResource` gRPC of Attestation Agent inside the same Pod. More details about `GetResource` and Attestation Agent please refer to [get-resource-service](ccv1_image_security_design.md#get-resource-service) and [attestation-agent](ccv1_image_security_design.md#attestation-agent).
+3. The `image-rs` module then match the entries inside the `auth.json` using longest prefix match. In this story, it will do:
+* Firstly look for the entry `private-registry.org/busybox`, but there is not.
+* Then search for the entry `private-registry.org`, and the matched value will be `"auth": "QWxpY2U6cHN3ZAo="`.
+In other scenarios, if no entries is matched, use `Anonymous` authentication in step 4.
+4. The `image-rs` module use the credential matched to pull the image.
+
+## Considerations
+
+### Where to store the `auth.json`?
+
+The same as `policy.json` in [image security](ccv1_image_security_design.md#policy).
+We put the file in `/run/image-security/auth.json` inside the Pod, because the `/run` directory is mounted in `tmpfs`, which is located in the encrypted memory protected by HW-TEE.
+
+### When `image-rs` should check and get `auth.json` if not present?
+
+There are two options:
```

Comment:
I wonder whether this document is a proposal or a design document. If it is the latter, should there be no multiple options and `may` statements in the document?

Reply:
Thanks, this document is a design document. I'll go to replace `may` statements. Besides, the reason why put options here is that this is not discussed before, and rationale of the choice should be declared.